CodeEditModules/Modules/AppPreferences/src/Model/General/GeneralPreferences.swift

```diff
@@ -127,12 +134,21 @@ public extension AppPreferences {
     /// The style for file extensions display
     ///  - **hideAll**: File extensions are hidden
     ///  - **showAll** File extensions are visible
-    ///  - **showOnly** Display specified file extensions
-    enum FileExtensions: String, Codable {
+    ///  - **showOnly** Specific file extensions are visible
+    enum FileExtensions: Codable, Hashable {
         case hideAll
         case showAll
         case showOnly
     }
+
+    /// The collection of displayed file extensions of ``FileExtensions/showOnly`` preference
+    struct ShowedFileExtensions: Codable, Hashable {
```

Comment:
I think `ShownFileExtensions` would be more appropriate than `ShowedFileExtensions`